polynote-frontend/polynote/ui/util/ui_event.ts

```diff
@@ -57,7 +72,12 @@ export class UIMessageTarget {
             }
         }
         if (this.uiParent instanceof UIMessageTarget) {
-            this.uiParent.publish(event.copy())
+            if (this.uiParent !== this) {
+                this.uiParent.publish(event.copy())
+            }
+        } else {
+            // this is for catching unintentional breaks in the chain.
+            console.warn("Called publish() on parent-less instance! This might be an accidental break in the chain. Otherwise, consider calling makeRoot() when initializing this UIMessageTarget")
```

Comment:
We should pay attention to this warning (I had initially made it throw but figured that might be a bit draconian) because it might indicate a problem that's pretty annoying to debug (a break in the chain of events bubbling up)